tests/test_cli_sync.py

```diff
@@ -111,42 +111,65 @@ def test_merge_error(runner):
 
 
 @pytest.mark.parametrize(
-    ("cli_flags", "expected_install_flags"),
+    "install_flags",
     [
-        (["--find-links", "./libs"], ["-f", "./libs"]),
-        (["--no-index"], ["--no-index"]),
-        (["--index-url", "https://example.com"], ["-i", "https://example.com"]),
-        (
-            ["--extra-index-url", "https://foo", "--extra-index-url", "https://bar"],
-            ["--extra-index-url", "https://foo", "--extra-index-url", "https://bar"],
-        ),
-        (
-            ["--trusted-host", "https://foo", "--trusted-host", "https://bar"],
-            ["--trusted-host", "https://foo", "--trusted-host", "https://bar"],
-        ),
-        (
-            ["--extra-index-url", "https://foo", "--trusted-host", "https://bar"],
-            ["--extra-index-url", "https://foo", "--trusted-host", "https://bar"],
-        ),
-        (["--user"], ["--user"]),
-        (["--cert", "foo.crt"], ["--cert", "foo.crt"]),
-        (["--client-cert", "foo.pem"], ["--client-cert", "foo.pem"]),
+        ["--find-links", "./libs1", "--find-links", "./libs2"],
+        ["--no-index"],
+        ["--index-url", "https://example.com"],
+        ["--extra-index-url", "https://foo", "--extra-index-url", "https://bar"],
+        ["--trusted-host", "foo", "--trusted-host", "bar"],
+        ["--user"],
+        ["--cert", "foo.crt"],
+        ["--client-cert", "foo.pem"],
     ],
 )
 @mock.patch("piptools.sync.check_call")
-def test_pip_install_flags(check_call, cli_flags, expected_install_flags, runner):
+def test_pip_install_flags(check_call, install_flags, runner):
     """
     Test the cli flags have to be passed to the pip install command.
     """
     with open("requirements.txt", "w") as req_in:
         req_in.write("six==1.10.0")
 
-    runner.invoke(cli, cli_flags)
+    runner.invoke(cli, install_flags)
 
     call_args = [call[0][0] for call in check_call.call_args_list]
-    assert [args[6:] for args in call_args if args[3] == "install"] == [
-        expected_install_flags
-    ]
+    called_install_options = [args[6:] for args in call_args if args[3] == "install"]
+    assert called_install_options == [install_flags], "Called args: {}".format(
+        call_args
+    )
+
+
+@pytest.mark.parametrize(
+    "install_flags",
+    [
+        ["--no-index"],
+        ["--index-url", "https://example.com"],
+        ["--extra-index-url", "https://example.com"],
+        ["--find-links", "./libs1"],
+        ["--trusted-host", "example.com"],
+        ["--no-binary", ":all:"],
+        ["--only-binary", ":all:"],
+    ],
+)
+@mock.patch("piptools.sync.check_call")
```

Comment:
FTR the preferred way of patching things in `pytest` is to use a built-in fixture called `monkeypatch`.

Reply:
Are there any links to the discussion about `monkeypatch` vs `mock`? I see only https://github.com/pytest-dev/pytest/issues/4576 and a few others where arguments are mostly based on personal preferences. Personally, I do like `@mock.patch` decorator.

Would you like to open an issue so that we can discuss it and decide whether it's worth to refactor codebase form mock to monkeypatch?

Reply:
I just think that it's cleaner because it's natively integrated + fewer deps.
Also, it automatically, unpatches things on exit.

Reply:
> fewer deps

MagicMock needs pytest-mock btw.

Reply:
I think it's vice versa

Reply:
How would you rewrite this test with `monkeypatching`? I really want to see the benefit of it (not trolling).

Reply:
So instead of a decorator inside of the test method, you add the `monkeypatch` fixture to its attrs and then do smth like
`monkeypatch.setattr('piptools.sync.check_call', <mock_obj_here>)`.

Reply:
How would you assert the calls?

Reply:
You cannot (easily).
monkeypatch is very basic, and you would use e.g. pytest-mock to get unittest's mock into pytest often - but can use it directly also.
I think monkeypatch is fine for simple things, but not for asserting calls etc.

Reply:
Can't you do that with `MagicMock`?

Reply:
Sure.. but that's not `monkeypatch`.

Reply:
Wait, the patching method may still be `monkeypatch` while the value would be `MagicMock()`.

Reply:
@webknjaz

> I just think that it's cleaner because it's natively integrated + fewer deps.

You are right, it could be the `mock.MagicMock`. Instead of using `mock` and patch things with `mock`, the suggestion to use `monkeypatch` with `mock.MonkeyPatch`. I don't get it, how it can be "fewer deps"?

Reply:
It's fewer deps under py3. Also, I'm not sure I'd actually endorse magic mock, better define a proper fake object which would fit expectations instead of implicitly hiding possibly malicious behaviors.

Reply:
I see. [Pretend](https://github.com/alex/pretend) might be an alternative of MagickMock, btw.